codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/RuntimeTypes.kt

```diff
@@ -23,26 +24,33 @@ import software.amazon.smithy.rust.codegen.rustlang.asType
 import software.amazon.smithy.rust.codegen.util.orNull
 import java.util.Optional
 
+private const val DEFAULT_KEY = "DEFAULT"
+
 /**
  * Location of the runtime crates (aws-smithy-http, aws-smithy-types etc.)
  *
- * This can be configured via the `runtimeConfig.version` field in smithy-build.json
+ * This can be configured via the `runtimeConfig.versions` field in smithy-build.json
  */
-data class RuntimeCrateLocation(val path: String?, val version: String?) {
+data class RuntimeCrateLocation(val path: String?, val versions: CrateVersionMap) {
     init {
-        check(path != null || version != null) {
-            "path ($path) or version ($version) must not be null"
+        check(path != null || versions.map.isNotEmpty()) {
+            "path ($path) or versions ($versions) must not be null or empty"
         }
     }
 
     companion object {
-        fun Path(path: String) = RuntimeCrateLocation(path, null)
+        fun Path(path: String) = RuntimeCrateLocation(path, CrateVersionMap(emptyMap()))
     }
 }
 
-fun RuntimeCrateLocation.crateLocation(): DependencyLocation = when (this.path) {
-    null -> CratesIo(this.version!!)
-    else -> Local(this.path, this.version)
+fun RuntimeCrateLocation.crateLocation(crateName: String?): DependencyLocation {
+    val version = crateName.let { versions.map[crateName] } ?: versions.map[DEFAULT_KEY]
+    return when (this.path) {
+        // CratesIo needs an exact version. However, for local crate we do not
+        // provide a detected version unless user explicitly sets via `DEFAULT` key.
+        null -> CratesIo(version ?: defaultRuntimeCrateVersion())
```

Comment:
I don't think this has been documented in the changelog (i.e., that when you don't specify anything in `versions`, then the latest smithy-rs version number is used for all crates).

Reply:
Fixed with b31e0b4